R/utils-json.R

```diff
@@ -263,10 +263,15 @@ subsetByMonth <- function(pts, onlyMonth) {
   }
   return(pts)
 }
+
 getYvals_approvals <- function(object, num_vals){
   ylim <- ylim(object)$side.2[1]
   yvals <- rep(ylim, num_vals)
-  return(yvals)
+  
+  # The constant scaling factor below was reverse-engineered from analyzing the 
+  # (pandoc?) SVG in Inkscape. I'm not entirely sure why it works, and it's easy
+  # to imagine why it wouldn't in some cases.
+  return(0.908196 * yvals)
```

Comment:
The function `rep` needs the value to repeat and then the number of times to repeat it. `num_vals` is the number of points (x values) that exist for approvals. We use this so that we create the exact same number of y values to go with it. 

The approval "line" is actually a rectangle now. `y0` is the lower y value of the rectangle and `y1` is the upper y value of the rectangle. Currently, `y0` is using the values from `getYvals_approvals()` (so the lowest data extent, not plot extent) and `y1` is that number plus something define in `addHeight()`. I'm proposing that `y0` should be the lowest plot extent (using `par("usr")` calculations), and `y1` (the upper part of the rectangle) exist at the lowest data extent. I'm not certain this will work, but that's what I imagine COULD work.

 `isLogged` is used to see if a specific timeseries is logged and if so, indicates that the plot y-axis should be logged. You'd need to check a bunch of the timeseries and determine if any are logged to use that function. The better way is to use either `primaryInfo$logAxis` (and `primaryInfo$isInverted` for reverse y axes) created in uvhydrograph-render ***OR*** inside the plot object (which is already passed into the `getYvals_approvals()` function) by checking `object$side.2$log` (and `object$side.2$reverse` for reversed y axes).